### PR TITLE
ESC-703 Nil Return Journey Changes

### PR DESCRIPTION
**Pull request by carwynellis**
Summary of changes
* revisions to the nil return flow which now includes a new confirmation page for submission of a nil return
* some minor refactoring and tidying up

### CODE REVIEW

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/NoClaimNotificationController.scala [hunk 1]
```diff
@@ -40,45 +44,97 @@ class NoClaimNotificationController @Inject() (
...
+
+          val lastSubmitted = undertakingSubsidies.lastSubmitted.orElse(undertaking.lastSubsidyUsageUpdt)
+
+          Ok(
+            noClaimNotificationPage(
+              noClaimForm,
+              previous,
+              undertakingSubsidies.hasNeverSubmitted, 
+              startDate.toDisplayFormat,
+              lastSubmitted.map(_.toDisplayFormat).getOrElse(""), // TODO - can we display something sensible if the value is missing?
```
Comment (sam-miele) [inline app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/NoClaimNotificationController.scala:70]: Quick question - was this meant to be fixed in this PR?

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/NoClaimNotificationController.scala [hunk 2]
```diff
@@ -40,45 +44,97 @@ class NoClaimNotificationController @Inject() (
...
+          noClaimForm
+            .bindFromRequest()
+            .fold(
+              errors => BadRequest(
+                noClaimNotificationPage(
+                  errors,
+                  previous,
+                  undertakingSubsidies.hasNeverSubmitted,
+                  startDate.toDisplayFormat,
+                  lastSubmitted.map(_.toDisplayFormat).getOrElse("") // TODO - can we display something sensible if the date is missing?
```
Comment (sam-miele) [inline app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/NoClaimNotificationController.scala:124]: same here.

**Review by sam-miele — APPROVED** [review]
LGTM - just checking those todo's weren't meant for now.

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/NoClaimNotificationController.scala (hunk 1 above)
Comment (carwynellis) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/NoClaimNotificationController.scala:70]: Not yet. Something to revisit later.

**Pull request merged**